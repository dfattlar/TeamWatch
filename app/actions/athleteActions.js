import * as types from './actionTypes';

export function addAthlete() {
    return {
        type: types.ADD_ATHLETE
    }
}

export function addAthleteToWatch(id, name) {
    return {
        type: types.ADD_ATHLETE_TO_WATCH,
        payload: { id,name }
    }
}

export function removeAthleteFromWatch(id) {
    return {
        type: types.REMOVE_ATHLETE_FROM_WATCH,
        payload: { id }
    }
}


export function addAthleteError() {
    return {
        type: types.ADD_ATHLETE_ERROR
    }
}

export function newAthleteInput(name) {
    return {
        type: types.NEW_ATHLETE_INPUT,
        newAthleteInput: name
    }
}
