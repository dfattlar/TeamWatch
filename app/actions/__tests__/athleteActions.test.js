import * as actions from '../athleteActions';

describe('Athlete Actions', () => {
  it('should create an action to input a new athlete', () => {
    const newAthleteInput = 'Athlete1';
    expect(actions.newAthleteInput(newAthleteInput)).toMatchSnapshot();
  });

  it('should create an action to add an athlete', () => {
    expect(actions.addAthlete()).toMatchSnapshot();
  });

  it('should create an action to add an athlete error', () => {
    expect(actions.addAthleteError()).toMatchSnapshot();
  });

  it('should create an action to add an athlete to the watch', () => {
    const id = 'id_123';
    const name = 'Athlete1';
    expect(actions.addAthleteToWatch(id, name)).toMatchSnapshot();
  });

  it('should create an action to remove an athlete from the watch', () => {
    const id = 'id_123';
    expect(actions.removeAthleteFromWatch(id)).toMatchSnapshot();
  });

  it('should create an action to delete an athlete', () => {
    const id = 'id_123';
    expect(actions.deleteAthlete(id)).toMatchSnapshot();
  });
});
