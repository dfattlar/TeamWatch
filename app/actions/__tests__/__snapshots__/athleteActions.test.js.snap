// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Athlete Actions should create an action to add an athlete 1`] = `
Object {
  "type": "ADD_ATHLETE",
}
`;

exports[`Athlete Actions should create an action to add an athlete error 1`] = `
Object {
  "type": "ADD_ATHLETE_ERROR",
}
`;

exports[`Athlete Actions should create an action to add an athlete to the watch 1`] = `
Object {
  "payload": Object {
    "id": "id_123",
    "name": "Athlete1",
  },
  "type": "ADD_ATHLETE_TO_WATCH",
}
`;

exports[`Athlete Actions should create an action to delete an athlete 1`] = `
Object {
  "id": "id_123",
  "type": "DELETE_ATHLETE",
}
`;

exports[`Athlete Actions should create an action to input a new athlete 1`] = `
Object {
  "newAthleteInput": "Athlete1",
  "type": "NEW_ATHLETE_INPUT",
}
`;

exports[`Athlete Actions should create an action to remove an athlete from the watch 1`] = `
Object {
  "id": "id_123",
  "type": "REMOVE_ATHLETE_FROM_WATCH",
}
`;
